source/slang/slang-emit-spirv.cpp

```diff
@@ -2954,9 +2954,24 @@ struct SPIRVEmitContext
             }
             break;
 
+        case kIROp_VulkanHitAttributesDecoration:
         case kIROp_VulkanCallablePayloadDecoration:
+        case kIROp_VulkanCallablePayloadInDecoration:
+            ensureExtensionDeclaration(UnownedStringSlice("SPV_KHR_ray_tracing"));
+            requireSPIRVCapability(SpvCapabilityRayTracingKHR);
+            goto rayExtensionAddition;
         case kIROp_VulkanHitObjectAttributesDecoration:
+            // needed since GLSL will not set optypes accordingly, but will keep the decoration 
+            ensureExtensionDeclaration(UnownedStringSlice("SPV_NV_shader_invocation_reorder"));
+            requireSPIRVCapability(SpvCapabilityShaderInvocationReorderNV);
+            goto rayExtensionAddition;
         case kIROp_VulkanRayPayloadDecoration:
+        case kIROp_VulkanRayPayloadInDecoration:
+            // needed since GLSL will not set optypes accordingly, but will keep the decoration 
+            ensureExtensionDeclaration(UnownedStringSlice("SPV_KHR_ray_query"));
+            requireSPIRVCapability(SpvCapabilityRayQueryKHR);
+            goto rayExtensionAddition;
+rayExtensionAddition:;
```

Comment:
Let's not use GOTO within switch statement.
In this case, you can set a temporary variable to be true and handle the action later.
```
bool isRayExtension = false;
...
case kIROp_VulkanCallablePayloadInDecoration:
...
    isRayExtension = true;
    break;
...
if (isRayExtension)
{
   emitOpDecorateLocation(...);
}
````
In general, "goto" should be limited to error handling; not for code reuse.